CodeEdit/Features/Documents/Controllers/CodeEditSplitViewController.swift

```diff
@@ -39,10 +44,12 @@ final class CodeEditSplitViewController: NSSplitViewController {
         fatalError("init(coder:) has not been implemented")
     }
 
-    // TODO: Set user preferences width if it is not the snap width
-//    override func viewWillAppear() {
-//        super.viewWillAppear()
-//    }
+    override func viewWillAppear() {
+        super.viewWillAppear()
+        let width = workspace.getFromWorkspaceState(key: self.widthStateName) as? CGFloat
+        splitView.setPosition(width ?? .snapWidth, ofDividerAt: .zero)
+        setWidthFromState = true
+    }
```

Comment:
Found a little problem...

As discussed in #976, we should initialize the sidebar with a width only if the user-preferred width exists and is not the snap width, otherwise, the haptic feedback for snapping the sidebar will be triggered. Here's a suggested solution:

```suggestion
	override func viewWillAppear() {
        super.viewWillAppear()
        let width = workspace.getFromWorkspaceState(key: self.widthStateName) as? CGFloat ?? .snapWidth
        if width != .snapWidth {
            splitView.setPosition(width, ofDividerAt: .zero)
            setWidthFromState = true
        }
    }
```

Reply:
When setting the sidebar width to the snap width programmatically, the haptic feedback should not be triggered.

Reply:
Yeah looks good. But I would add comments/documentation on why it is written this way, for future developers that read this code.

Reply:
> Yeah looks good. But I would add comments/documentation on why it is written this way, for future developers that read this code.

I added some comments about what has been done here.

> When setting the sidebar width to the snap width programmatically, the haptic feedback should not be triggered.

My idea would be to override `setPosition` (or even extend `NSSplitView`) and have a boolean indicating if the change was made programmatically, I don't think that this is the best solution though, and I don't have other solutions for now... I would also try to find other solutions for it.